packages/lib-react-components/src/FavouritesButton/FavouritesButton.spec.js

```diff
@@ -1,100 +1,76 @@
-      wrapper.simulate('click')
-      const icon = wrapper.prop('icon')
-      expect(icon.props.fill).to.equal('none')
+    it('should toggle favourites on then off', async function () {
+      await user.click(screen.getByRole('checkbox', { checked: false}))
+      await user.click(screen.getByRole('checkbox', { checked: true }))
+      expect(screen.getByRole('checkbox', { checked: false })).to.exist()
+      expect(screen.getByText('Add to favorites')).to.exist()
     })
+  })
+
+  describe('when favourited', function () {
+    const FavouritedStory = composeStory(Favourited, Meta)
 
-    it('should call props.onClick', function () {
-      wrapper.simulate('click')
-      expect(onClickStub).to.have.been.calledOnce()
+    beforeEach(function () {
+      render(<FavouritedStory />)
     })
-  })
 
-  describe('when checked', function () {
-    before(function () {
-      wrapper = shallow(<FavouritesButton theme={mockTheme} checked />)
+    it('should show a favourited icon', function () {
+      expect(screen.getByRole('checkbox', { checked: true })).to.exist()
+      expect(screen.getByText('Added to favorites')).to.exist()
     })
 
-    it('should display a filled icon', function () {
-      const button = wrapper.find(MetaToolsButton)
-      const { icon } = button.props()
-      const fill = mockTheme.global.colors.statusColors.error
-      expect(icon).to.deep.equal(<HeartIcon color='dark-5' fill={fill} size='15px' />)
+    it('should toggle favourites off', async function () {
+      await user.click(screen.getByRole('checkbox', { checked: true}))
+      expect(screen.getByRole('checkbox', { checked: false })).to.exist()
+      expect(screen.getByText('Add to favorites')).to.exist()
     })
 
-    it('should be checked', function () {
-      const checked = wrapper.find(MetaToolsButton).prop('aria-checked')
-      expect(checked).to.be.true()
+    it('should toggle favourites off then on', async function () {
+      await user.click(screen.getByRole('checkbox', { checked: true}))
+      await user.click(screen.getByRole('checkbox', { checked: false }))
+      expect(screen.getByRole('checkbox', { checked: true })).to.exist()
+      expect(screen.getByText('Added to favorites')).to.exist()
     })
   })
 
   describe('when disabled', function () {
-    const onClick = sinon.stub()
-    wrapper = shallow(
-      <FavouritesButton
-        disabled
-        onClick={onClick}
-      />
-    )
+    const DisabledStory = composeStory(Disabled, Meta)
+
+    beforeEach(function () {
+      render(<DisabledStory />)
+    })
 
-    it('should not be clickable', function () {
-      wrapper.find(MetaToolsButton).simulate('click')
-      expect(onClick).to.not.have.been.called()
+    it('should not be clickable', async function () {
+      expect(screen.getByRole('checkbox', { checked: false })).to.exist()
+      expect(screen.getByText('Add to favorites')).to.exist()
+      await user.click(screen.getByRole('checkbox', { checked: false}))
```

Comment:
Weird that this doesn't throw. Disabled inputs aren't focusable and can't be clicked.